src/components/executive-education-2u/ExecutiveEducation2UError.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ArrowBack } from '@edx/paragon/icons';
+import { Button, Col, Row } from '@edx/paragon';
+import {
+  ErrorPage,
+} from '../error-page';
+
+const ExecutiveEducation2UErrorPage = ({ failureReason, httpReferrer }) => {
+  const createExecutiveEducationFailureMessage = (failureCode) => {
+    const failureCodeMessages = {
+      no_offer_available: 'No offer is available to cover this course.',
+      no_offer_with_enough_balance: 'Your organization doesn’t have sufficient balance to cover this course.',
+      no_offer_with_enough_user_balance: 'You don’t have sufficient balance to cover this course.',
+      system_error: 'System Error has occured.',
+      default: 'An error has occured.',
+    };
+    return failureCodeMessages[failureCode] ? failureCodeMessages[failureCode] : failureCodeMessages.default;
+  };
+  return (
```

Comment:
Did we also need to include the conditional rendering based on whether a `failureReason` is provided, e.g.:

```jsx
if (!failureReason) {
  return null;
}
```

Reply:
Ignore this. Looks the conditional rendering is handled by the parent now, which is fine! :)